### PR TITLE
Added image and version variable & resource limits

### PR DESCRIPTION
**Pull request by martijnvdp**
minor changes: 
- Changed token var names to pass tfsec security check
- Added image and version variables so an image on ecr can be used
- enable/added vars to set resource limits
- added option to pass extra resources to the rbac profile of the service account

### CODE REVIEW

**Comment by redeux** [comment]
Thanks for the contribution! I looked over the changes briefly, and they make sense to me. Let me know when this is ready for a formal review.

**Comment by martijnvdp** [comment]
yes this is ready, maybe the default resource values could still be improved , havent put much time in that.

File: kubernetes_deployment.tf [hunk 1]
```diff
@@ -11,7 +11,7 @@ resource "kubernetes_deployment" "tfc_agent" {
   }
 
   spec {
-    replicas = 1
+    replicas = var.agent_replicacount
```
Comment (redeux) [inline kubernetes_deployment.tf:14]: should this be `agent_replicas` to match the manifest attribute name?

File: variables.tf [hunk 2]
```diff
@@ -20,15 +16,26 @@ variable "agent_log_level" {
...
-  type        = bool
-  default     = true
-  description = "Agents will self-update if set to false."
+variable "agent_name" {
+  type        = string
+  description = "The TFC agent token description defined in TFC at app/<org>/settings/agents."
 }
 
-variable "namespace" {
+variable "agent_replicacount" {
```
Comment (redeux) [inline variables.tf:24]: Same question about `replicas`

File: variables.tf [hunk 3]
```diff
@@ -41,4 +48,40 @@ variable "create_namespace" {
...
+
+variable "limits_cpu" {
+  type        = string
+  default     = "1"
+  description = "CPU hard limits."
+}
+
+variable "limits_memory" {
+  type        = string
+  default     = "2Gi"
```
Comment (redeux) [inline variables.tf:61]: I'm worried that this is too low.  I'm going to see if I can find sensible requests and limits.

**Comment by redeux** [comment]
Thanks for the contribution!  I left just a couple comments.  I'm also going to see if I can find the appropriate specs for the requests and limits.

**Comment by redeux** [comment]
It looks like the default limits you put are pretty sensible.

File: kubernetes_deployment.tf (hunk 1 above)
Comment (martijnvdp) [reply kubernetes_deployment.tf]: yes that might be better i've changed it

**Comment by martijnvdp** [comment]
Thanks for the review , i've renamed the variable and also added cilium resources to the rbac to be able to deploy cilium network policys with terraform using the agent.

**Comment by redeux** [comment]
This mostly looks good to me.  A couple more changes I would request.

- Changing the variable names makes this a breaking change, so the version bump will be to `0.1.0`
- I'm not in favor of including custom objects in the RBAC policy. I think we can use the [concat function](https://www.terraform.io/docs/language/functions/concat.html) to merge the standard list with any user defined additions.
- I noticed the `agent_token` variable isn't marked sensitive, but it probably should be.

**Review by redeux — CHANGES_REQUESTED** [review]
This mostly looks good to me.  A few more changes I would request.

- Changing the variable names makes this a breaking change, so the version bump will be to `0.1.0`
- I'm not in favor of including custom objects in the RBAC policy. I think we can use the [concat function](https://www.terraform.io/docs/language/functions/concat.html) to merge the standard list with any user defined additions.
- I noticed the `agent_token` variable isn't marked sensitive, but it probably should be.

**Comment by martijnvdp** [comment]
> This mostly looks good to me. A few more changes I would request.
> 
> * Changing the variable names makes this a breaking change, so the version bump will be to `0.1.0`
ok I've changed the version to 0.1.0
> * I'm not in favor of including custom objects in the RBAC policy. I think we can use the [concat function](https://www.terraform.io/docs/language/functions/concat.html) to merge the standard list with any user defined additions.
Yes that makes sense , i've changed this and added optional variables to add additional resources/api-groups
> * I noticed the `agent_token` variable isn't marked sensitive, but it probably should be.
Yes you're right should be sensitive i've changed it
Thanks for the review

**Review by redeux — COMMENTED** [review]
The module apply fails if you leave `cluster_access_rbac_api_gourps` or `cluster_acess_rbac_resources` empty.  Looks like everything works fine if the default is changed to `  default     = []`

**Pull request merged**